### PR TITLE
METRON-2036 Maven builds fail locally in HDFSWriterTest

### PR DESCRIPTION
**Pull request by mmiklavc**
## Contributor Comments

See full context and description here - https://issues.apache.org/jira/browse/METRON-2036

**Testing**

1. Travis should succeed
2. Run mvn clean install from your local CLI. HdfsWriterTest should pass.
3. Run HdfsWriterTest from your IDE (I am running IntelliJ). All tests should pass there as well.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- n/a If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- n/a Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- n/a Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -38,6 +38,24 @@
...
+
+        * However, if the JAR-class-path points to a JAR file that was already searched (for example, an extension, or a
+          JAR file that was listed earlier in the class path) then that JAR file will not be searched again. (This
+          optimization improves efficiency and prevents circular searches.) Such a JAR file is searched at the point that it
+          appears, earlier in the class path.
+
+        * If a JAR file is installed as an extension, then any JAR-class-path it defines is ignored. All the classes
+          required by an extension are presumed to be part of the SDK or to have themselves been installed as extensions.
+    -->
+
```
Comment (ottobackwards) [inline pom.xml]: On of the benefits of the NAR/Bundles was handling this

**Comment by ottobackwards** [comment]
+1, can you create a jira to upgrade when the new release comes out, and to remove this change?  Link it to the jira this refers to

**Comment by nickwallen** [comment]
Why did this never fail in Travis?

**Comment by mmiklavc** [comment]
@ottobackwards - Ticket for upgrading JUnit and removing this fix -> https://issues.apache.org/jira/browse/METRON-2037

@nickwallen - I don't have a satisfying answer, unfortunately. I suspect it may have to do with our running with multiple cores via `-T 2C` in Travis, which may serendipitously provide the writer test a fresh thread that has not been interrupted, but I can't say for certain without diving into JUnit and Maven Surefire plugin code.

**Comment by nickwallen** [comment]
I don't think we need to know the answer to that question for this PR to be merged.   As long as what you have works reliability, I am happy with the fix.

Food for thought, locally I am able to reliably fail the tests no matter how many threads I run the tests with `mvn test -T2C`.

**Comment by mmiklavc** [comment]
What version of Maven are you running?

```
mvn -version
Apache Maven 3.5.2 (138edd61fd100ec658bfa2d307c43b76940a5d7d; 2017-10-18T01:58:13-06:00)
```

**Comment by mmiklavc** [comment]
Travis is using Maven 3.3.9 - https://travis-ci.org/apache/metron/jobs/506242356#L501. I would expect variability more from Surefire version differences, however, and I believe we specify explicitly in our poms.

**Comment by nickwallen** [comment]
I run mvn 3.3.9
```
$ mvn -version
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T11:41:47-05:00)
```

**Comment by mmiklavc** [comment]
Is anyone opposed to merging this now a few hours shy of 24, considering the implications to local development and testing?

**Comment by ottobackwards** [comment]
No, go ahead

**Pull request closed without merging**